timesketch/app.py

```diff
@@ -145,7 +147,20 @@ def create_app(config=None, legacy_ui=False):
     api_v1 = Api(app, prefix="/api/v1")
     for route in V1_API_ROUTES:
         api_v1.add_resource(*route)
+    
+    # Returns 404 for invalid api routes
+    @app.route('/api/v1/<path:path>')
+    @login_required
+    def handle_invalid_route(path):
+        """If a request hits this route, that requested endpoint does not exist.
 
+        Returns:
+            HTTP response object (instance of jsonify)        
+        """
```

Comment:
```suggestion
        """Error handler for non-existent API routes.

        Raises:
            ApiHTTPError: Error 404 - not found        
        """
```

Updating the doc string to reflect the changes.